[DEPRECATIONS] Remove EUI Charts, Sass deprecations

snide opened the pull request:
### Summary

Following our deprecation schedule https://github.com/elastic/eui/issues/1469 this removes the Eui Charts set of components and docs. I did the following

* Removed the components and docs for EUI charts
* Removed all references to `experimental` folders and code (since we don't use it anymore)
* Removed `react-vis` from our deps
* Remove the `k6` theme and dist output
* In addition, I removed all Sass deprecations we had listed in the Sass codebase. Most of them were over 6 months old.

We can't merge this till @simianhacker removes `EuiSeriesChart` usage in the infra-ui, but he's promised soonish there. Until that's ready I'll leave this in draft mode. 

### Checklist

- [ ] ~This was checked in mobile~
- [ ] ~This was checked in IE11~
- [x] This was checked in dark mode
- [ ] ~Any props added have proper autodocs~
- [ ] ~Documentation examples were added~
- [x] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately
- [x] This was checked for breaking changes and labeled appropriately
- [x] Jest tests were updated or added to match the most common scenarios
- [ ] ~This was checked against keyboard-only and screenreader scenarios~
- [ ] ~This required updates to Framer X components~

cchaos commented:
Maybe we should also merge this on a fresh master (after a release) so it doesn't get mixed up with other bug fixes etc..

cchaos commented:
Can we also deprecate the K6 theme in the theme switcher?

<img src="https://d.pr/free/i/WDA5P5+" width="50%" />

snide commented:
> Maybe we should also merge this on a fresh master (after a release) so it doesn't get mixed up with other bug fixes etc..

I think I'll have to keep this up to date with rebases for a week or so. We can't merge till @simianhacker is done in kibana. I just wanted to get it going to provide more imminent awareness of its arrival :) In fact, we should let @elastic/cloud-ui know and make sure they're not using the charts either. 

> Can we also deprecate the K6 theme in the theme switcher?

Yep. I'll take care of it.

andrew-moldovan commented:
Yep, we're not using any charts from EUI

thompsongl commented:
I know this is still in draft, but we'll want to make sure to run `yarn` and commit the resulting `yarn.lock` when ready. I expect we'll see a big diff there.

snide commented:
> I know this is still in draft, but we'll want to make sure to run yarn and commit the resulting yarn.lock when ready. I expect we'll see a big diff there.

Yep, purposely didn't run it in the draft since I can better watch when i need to rebase (yarn.lock changes a bunch / will always conflict)

I'll add some todos as a reminder though.

cristina-eleonora commented:
Will these charts be replaced with any new ones, or are we removing the concept of EUI charts entirely?

cchaos commented:
@cristina-eleonora The EUI charts are being deprecated in favor of using [elastic-charts](https://elastic.github.io/elastic-charts) which we will be adding a few examples and documentation to EUI as well.

cristina-eleonora commented:
Awesome, thanks @cchaos !

snide commented:
Thanks to @simianhacker and https://github.com/elastic/kibana/pull/41262 we're ready to merge this into EUI. I updated the `yarn.lock` and removed all references to the `k6` theme. Note that I see a couple spots in Kibana that are pulling in the `k6` JSON files (looks like within beats), so we'll want to adjust those to point to the EUI one on our next Kibana update.

This is ready for review.

chandlerprall commented on CHANGELOG.md:
```diff
@@ -4,6 +4,12 @@
 
 - Fixed `EuiComboBox`'s padding on the right ([#2135](https://github.com/elastic/eui/pull/2135))
 
+**Deprecations**
```
This should be a different heading, as we aren't adding deprecations but removing functionality. Strictly speaking, this should be a breaking change.

snide commented:
@chandlerprall @thompsongl. Good point. Updated.
